pkg/piecestore/rpc/server/agreementsender/agreementsender.go

```diff
@@ -0,0 +1,151 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package agreementsender
+
+import (
+	"flag"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/zeebo/errs"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+
+	"storj.io/storj/pkg/node"
+	"storj.io/storj/pkg/overlay"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/piecestore/rpc/server/psdb"
+	"storj.io/storj/pkg/provider"
+	"storj.io/storj/pkg/utils"
+)
+
+var (
+	defaultCheckInterval = flag.Duration("piecestore.agreementsender.check_interval", time.Hour, "number of seconds to sleep between agreement checks")
+	defaultOverlayAddr   = flag.String("piecestore.agreementsender.overlay_addr", "127.0.0.1:7777", "Overlay Address")
+
+	// ASError wraps errors returned from agreementsender package
+	ASError = errs.Class("agreement sender error")
+)
+
+// AgreementSender maintains variables required for reading bandwidth agreements from a DB and sending them to a Payers
+type AgreementSender struct {
+	DB       *psdb.DB
+	overlay  overlay.Client
+	identity *provider.FullIdentity
+	errs     []error
+	mu       sync.Mutex
+}
+
+// Initialize the Agreement Sender
+func Initialize(DB *psdb.DB, identity *provider.FullIdentity) (*AgreementSender, error) {
+	overlay, err := overlay.NewOverlayClient(identity, *defaultOverlayAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	return &AgreementSender{DB: DB, identity: identity, overlay: overlay}, nil
+}
+
+// Run the afreement sender with a context to cehck for cancel
+func (as *AgreementSender) Run(ctx context.Context) error {
+	log.Println("AgreementSender is starting up")
```

Comment:
Shouldn't logging be done with `zap` logger?

Reply:
switched to zap